flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PartitionData.java

```diff
@@ -45,6 +51,14 @@
 
 	public abstract boolean isBuffer();
 
+	/**
+	 * Returns the buffer-format partition data with the provided memory segment or not.
+	 *
+	 * @param segment it might be needed to read the partition data into.
+	 * @return the buffer represents the partition data.
+	 */
+	public abstract Buffer getBuffer(@Nullable MemorySegment segment) throws IOException;
```

Comment:
TBH, I am a bit torn and not satisfied with introducing this method, since the argument `segment` seem not very elegant from the caller usage. But I did not think of a better option for resolving it now. :(